lib/pulse/lib/Pulse.Lib.AVLTree.fst

```diff
@@ -796,105 +826,133 @@ ensures is_tree tree l ** pure (y == T.tree_max l)
   }
 }
 
-
-
 fn rec delete_avl (#t:Type0) (cmp: T.cmp t) (tree:tree_t t) (key: t)
-requires is_tree tree 'l
-returns y:tree_t t 
-ensures (is_tree y (T.delete_avl cmp 'l key))
+  requires is_tree tree 'l
+  returns  y : tree_t t
+  ensures  is_tree y (T.delete_avl cmp 'l key)
 {
-  match tree{
-   None -> {
-    is_tree_case_none tree;
+  match tree {
+    None -> {
+      is_tree_case_none None;
+      rewrite is_tree None 'l as is_tree tree 'l;
```

Comment:
This is a common pattern in matches now.  Can you do a `rewrite each tree to None` instead?